### PR TITLE
Baseflight pid (pid_controller=2) horizon mode tuneup

### PR DESCRIPTION
**Pull request by nebbian**
This code uses a stick position algorithm for calculating the leveling force when inverted.  This is based on RC911's "Stick Mixing" algorithm on the KK2 controller, and is similar to the horizon mode in PID controller 0.  

Flight test is here:
https://www.youtube.com/watch?v=C_vZDNst6pM

When tuning, the Level P term is the leveling force when in angle mode.  The Level I term is the leveling force when in horizon mode, and the Level D term is the percentage of stick travel where leveling is active.  The default values seem to work fine.

### CODE REVIEW

File: src/main/flight/flight.c [hunk 1]
```diff
@@ -96,14 +96,38 @@ static void pidBaseflight(pidProfile_t *pidProfile, controlRateConfig_t *control
...
+        // Figure out the raw stick positions
+        stickPosAil = getRcStickPosition(FD_ROLL);
+        stickPosEle = getRcStickPosition(FD_PITCH);
+
+        if(abs(stickPosAil) > abs(stickPosEle)){
+            mostDeflectedPos = abs(stickPosAil);
+        }
+        else {
+            mostDeflectedPos = abs(stickPosEle);
+        }
```
Comment (ledvinap) [inline src/main/flight/flight.c]: Using single max() would be equally readable ;-)

`mostDeflectedPos = max(abs(getRcStickPosition(FD_ROLL)), abs(getRcStickPosition(FD_PITCH)));`

File: src/main/io/serial_msp.c [hunk 2]
```diff
@@ -846,7 +846,7 @@ static bool processOutCommand(uint8_t cmdMSP)
                 if (i == PIDLEVEL) {
                     serialize8(constrain(lrintf(currentProfile->pidProfile.A_level * 10.0f), 0, 250));
                     serialize8(constrain(lrintf(currentProfile->pidProfile.H_level * 10.0f), 0, 250));
-                    serialize8(0);
+                    serialize8(constrain(lrintf(currentProfile->pidProfile.H_sensitivity), 0, 250));
```
Comment (ledvinap) [inline src/main/io/serial_msp.c]: no lrintf() is necessary,  H_sensitivity is uint8 ...

File: src/main/mw.c [hunk 3]
```diff
@@ -338,6 +338,11 @@ void mwArm(void)
     }
 }
 
+int32_t getRcStickPosition(int32_t axis) {
+
+    return min(abs(rcData[axis] - masterConfig.rxConfig.midrc), 500);
```
Comment (ledvinap) [inline src/main/mw.c]: IMO abs() is bad idea here - name it `getRcStickDeflection` or return negative value ...

**Comment by hydra** [comment]
We should probably rename this pid controller to something else too, since it never actually made its way into baseflight.

**Comment by hydra** [comment]
Ok, gonna merge this now.

**Comment by nebbian** [comment]
Nice work @hydra, I like the changes you made to this as well.  Will flight test the latest code this weekend.

**Pull request merged**